File: C#/InterviewCampProblems/InterviewCampProblems.Test/UnitTest4.cs
/*
Given an array, reverse the order of its elements.For example, [3,5,2,5,2,3,9] → [9,3,2,5,2,5,3]
*/

using System;
using InterviewCampProblems.Code;
using Xunit;

namespace InterviewCampProblems.Test {
    public class UnitTest4 {
        [Fact]
        public void Test1 () {

            int[] expected = new int[] { 9, 3, 2, 5, 2, 5, 3 };
            int[] arr = new int[] { 3, 5, 2, 5, 2, 3, 9 };
            var actual = Class4.ReverseElement (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test2 () {

            int[] expected = null;
            int[] arr = null;
            var actual = Class4.ReverseElement (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test3 () {

            int[] expected = new int[] { };
            int[] arr = new int[] { };
            var actual = Class4.ReverseElement (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test4 () {

            int[] expected = new int[] { 1 };
            int[] arr = new int[] { 1 };
            var actual = Class4.ReverseElement (arr);
            Assert.Equal (expected, actual);
        }
    }
}

File: Java/InterviewCamp/src/test/java/UnitTest4.java
//Given an array, reverse the order of its elements.For example, [3,5,2,5,2,3,9] → [9,3,2,5,2,5,3]

import org.junit.Assert;
import org.junit.Test;

public class UnitTest4 {

    @Test
    public void When_Input_Correct() {

        //Arrange
        int[] arr = new int[] {1,2,3,4,5};
        int[] outputarr = new int[] {5,4,3,2,1};
        //Act
        int[] output = Class4.ReverseArray(arr);

        //Assert
        Assert.assertArrayEquals(output,outputarr);

    }
}